crypto/dsa/dsa_ameth.c

```diff
@@ -514,6 +521,46 @@ static int dsa_pkey_ctrl(EVP_PKEY *pkey, int op, long arg1, void *arg2)
 
 }
 
+static size_t dsa_pkey_dirty_cnt(const EVP_PKEY *pkey)
+{
+    return pkey->pkey.dsa->dirty_cnt;
+}
+
+static void *dsa_pkey_export_to(const EVP_PKEY *pk, EVP_KEYMGMT *keymgmt)
+{
+    DSA *dsa = pk->pkey.dsa;
+    OSSL_PARAM_BLD tmpl;
+    const BIGNUM *p = DSA_get0_p(dsa), *g = DSA_get0_g(dsa), *q = DSA_get0_q(dsa);
+    const BIGNUM *pub_key = DSA_get0_pub_key(dsa);
+    const BIGNUM *priv_key = DSA_get0_priv_key(dsa);
+    OSSL_PARAM *params;
+    void *provkey = NULL;
+
+    if (p == NULL || q == NULL || g == NULL || pub_key == NULL)
```

Comment:
Not `pub_key`, as this may be just the domain parameters, which should be perfectly ok.
See https://github.com/openssl/openssl/pull/9670/files#diff-cd5a6dfdd711380eaa038ea0d9a0959dR540-R544

Reply:
Fixed.